pkg/build/api/validation/validation.go

```diff
@@ -171,6 +178,10 @@ func validateSource(input *buildapi.BuildSource, isCustomStrategy, isDockerStrat
 		}
 	}
 
+	if input.Git == nil && isJenkinsPipelineStrategyFromRepo {
+		allErrs = append(allErrs, field.Invalid(fldPath.Child("git"), "", "must be set when using Jenkins Pipeline strategy with Jenkinsfile from a git repo"))
```

Comment:
will JenkinsPipeline always require GIT repo? IOW. I thought we discussed that having Jenkinsfile from URL will not be allowed (so I would remove the second check).